flink-connectors/flink-hadoop-compatibility/src/main/java/org/apache/flink/api/java/hadoop/mapred/wrapper/HadoopInputSplit.java

```diff
@@ -103,23 +109,28 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
 		// read the parent fields and the final fields
 		in.defaultReadObject();
 
-		// the job conf knows how to deserialize itself
-		jobConf = new JobConf();
-		jobConf.readFields(in);
-
 		try {
 			hadoopInputSplit = (org.apache.hadoop.mapred.InputSplit) WritableFactories.newInstance(splitType);
-		}
-		catch (Exception e) {
+		} catch (Exception e) {
 			throw new RuntimeException("Unable to instantiate Hadoop InputSplit", e);
 		}
 
-		if (hadoopInputSplit instanceof Configurable) {
-			((Configurable) hadoopInputSplit).setConf(this.jobConf);
-		}
-		else if (hadoopInputSplit instanceof JobConfigurable) {
-			((JobConfigurable) hadoopInputSplit).configure(this.jobConf);
+		if (isConfigurable(hadoopInputSplit)) {
+			// the job conf knows how to deserialize itself
+			jobConf = new JobConf();
+			jobConf.readFields(in);
+
+			if (hadoopInputSplit instanceof Configurable) {
+				((Configurable) hadoopInputSplit).setConf(this.jobConf);
```

Comment:
Is it ok to `conf` this `hadoopInputSplit` before `hadoopInputSplit.readFields(in)`? Looks dangerous to me.

Reply:
The previous code is set conf first. But I think `setConf` after `readFields` is dangerous too.

Reply:
After I took a look to `org.apache.hadoop.hive.ql.io.HiveInputFormat.HiveInputSplit`, I think it is right to `setConf` before `readFields`, because in this class, its `readFields` depends on `conf`.